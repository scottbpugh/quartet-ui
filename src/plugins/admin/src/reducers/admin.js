export const initialData = () => {
  return {};
};
export default {};
