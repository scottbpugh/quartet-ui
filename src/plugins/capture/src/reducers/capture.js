// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {handleActions} from "redux-actions";
import {showMessage} from "lib/message";
import {pluginRegistry} from "plugins/pluginRegistration";
import actions from "../actions/capture";
import serverActions from "actions/serversettings";

export const initialData = () => ({
  servers: {}
});

export const loadRules = server => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.capture.capture_rules_list().then(result => {
          // load steps, all steps for all rules.
          // This may become an issue in the future, if so, a new backend API endpoint
          // needs to be added.
          client.apis.capture.capture_steps_list().then(steps => {
            let rules = result.body.map(rule => {
              // add steps to the rule.
              rule.steps = steps.body.filter(step => {
                if (step.rule === rule.id) {
                  return true;
                }
                return false;
              });
              return rule;
            });
            dispatch({
              type: actions.loadRules,
              payload: {
                serverID: server.serverID,
                rules: result.body
              }
            });
          });
        });
      });
  };
};

export const deleteRule = (server, rule) => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.capture.capture_rules_delete(rule).then(result => {
          return dispatch(loadRules(server));
        });
      });
  };
};

export const loadTasks = server => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.capture.capture_tasks_list().then(result => {
          dispatch({
            type: actions.loadTasks,
            payload: {
              serverID: server.serverID,
              tasks: result.body
            }
          });
        });
      });
  };
};

export const deleteStep = (server, step) => {
  return dispatch => {
    pluginRegistry
      .getServer(server.serverID)
      .getClient()
      .then(client => {
        client.apis.capture.capture_steps_delete(step).then(result => {
          return dispatch(loadRules(server));
        });
      });
  };
};

export default handleActions(
  {
    [actions.loadRules]: (state, action) => {
      if (!state.servers) {
        // temporary debug fix.
        state.servers = {};
      }
      return {
        ...state,
        servers: {
          ...state.servers,
          [action.payload.serverID]: {
            ...state.servers[action.payload.serverID],
            rules: action.payload.rules
          }
        }
      };
    },
    [actions.loadTasks]: (state, action) => {
      if (!state.servers) {
        // temporary debug fix.
        state.servers = {};
      }
      return {
        ...state,
        servers: {
          ...state.servers,
          [action.payload.serverID]: {
            ...state.servers[action.payload.serverID],
            tasks: action.payload.tasks
          }
        }
      };
    },
    [serverActions.serverUpdated]: (state, action) => {
      // we want to reload pools when new server is saved.
      /*action.asyncDispatch(
        loadPools(pluginRegistry.getServer(action.payload))
      );*/
      return {
        ...state,
        servers: {
          ...state.servers,
          [action.payload.serverID]: {}
        }
      };
    }
  },
  {}
);
