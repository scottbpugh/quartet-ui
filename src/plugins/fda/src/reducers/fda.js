// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
import actions from "../actions/fda";
const {handleActions} = qu4rtet.require("redux-actions");
const {setServerState} = qu4rtet.require("./lib/reducer-helper");
const {showMessage} = qu4rtet.require("./lib/message");
const {pluginRegistry} = qu4rtet.require("./plugins/pluginRegistration");

export const initialData = () => {
  return {
    servers: {}
  };
};

export const searchFDA = (server, searchField, lookup, skip) => {
  return dispatch => {
    fetch(
      `https://api.fda.gov/drug/label.json?search=${searchField}:${lookup}&limit=20&skip=${skip}`
    )
      .then(response => {
        if (!response.ok) {
          throw Error(response);
        }
        return response.json();
      })
      .then(data => {
        if (data.results && data.meta) {
          return dispatch({
            type: actions.loadFDAResults,
            payload: {
              serverID: server.serverID,
              searchField: searchField,
              lookup: lookup,
              items: data.results,
              skip: data.meta.results.skip,
              total: data.meta.results.total
            }
          });
        }
      })
      .catch(error => {
        return dispatch({
          type: actions.loadFDAResults,
          payload: {
            serverID: server.serverID,
            searchField: searchField,
            lookup: lookup,
            items: [],
            skip: 0,
            total: 0
          }
        });
        console.log(error);
      });
  };
};

export default handleActions(
  {
    [actions.loadFDAResults]: (state, action) => {
      return setServerState(state, action.payload.serverID, {
        fdaSearchField: action.payload.searchField,
        fdaLookup: action.payload.lookup,
        fdaItems: action.payload.items,
        fdaSkip: action.payload.skip,
        fdaTotal: action.payload.total
      });
    }
  },
  {}
);
