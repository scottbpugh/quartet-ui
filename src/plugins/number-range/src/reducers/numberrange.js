// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {handleActions} from "redux-actions";
import actions from "../actions/pools";
import {getPools, getRegion, getRegions, allocate} from "../lib/serialbox-api";
import {showMessage} from "../../../../lib/message";
import serverActions from "../../../../actions/serversettings";

export const initialData = () => ({
  servers: {},
  region: {},
  currentRegions: []
});

export const loadPools = server => {
  return dispatch => {
    getPools(server).then(pools => {
      dispatch({
        type: actions.loadPools,
        payload: {
          [server.serverID]: {pools: pools, server: server}
        }
      });
    });
  };
};

export const loadRegion = (server, regionName) => {
  return dispatch => {
    getRegion(server, regionName).then(region => {
      dispatch({
        type: actions.loadRegion,
        payload: region
      });
    });
  };
};

export const loadRegions = (server, pool) => {
  return dispatch => {
    getRegions(server, pool).then(regions => {
      console.log("regions", regions, "type of", typeof regions);
      dispatch({
        type: actions.loadRegions,
        payload: regions
      });
    });
  };
};

export const setAllocation = (server, pool, value) => {
  return dispatch => {
    allocate(server, pool, value).then(data => {
      // let's take a look at the data.
      if (data.detail) {
        // looks like an error.
        showMessage({type: "error", msg: data.detail});
      } else if (data.fulfilled === true) {
        showMessage({
          type: "success",
          msg: `${data.size_granted} allocated to region ${
            data.region
          }. Range is  ${data.numbers}`
        });
      }
      dispatch({type: actions.allocate, payload: data});
      // reload regions.
      getRegions(server, pool).then(regions => {
        dispatch({
          type: actions.loadRegions,
          payload: regions
        });
      });
    });
  };
};

export default handleActions(
  {
    [actions.loadPools]: (state, action) => {
      return {
        ...state,
        servers: {
          ...state.servers,
          ...action.payload
        }
      };
    },
    [actions.loadRegion]: (state, action) => {
      return {
        ...state,
        region: action.payload
      };
    },
    [actions.loadRegions]: (state, action) => {
      return {
        ...state,
        currentRegions: action.payload
      };
    },
    [actions.allocate]: (state, action) => {
      return {
        ...state,
        allocationDetail: action.payload
      };
    },
    [serverActions.serverUpdated]: (state, action) => {
      // we want to reload pools when new server is saved.
      console.log(
        "SERVER ADDED OR UPDATED. Number Range will fetch data on this server."
      );
      action.asyncDispatch(loadPools(action.payload));
      return {
        ...state
      };
    }
  },
  {}
);
