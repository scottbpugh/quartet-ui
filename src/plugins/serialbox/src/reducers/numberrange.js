// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {
    getPools,
    getRegion,
    getRegions,
    allocate,
    deleteRegion,
    deletePool
} from "../lib/serialbox-api";
import actions from "../actions/pools";

const {handleActions} = qu4rtet.require("redux-actions");
const {showMessage} = qu4rtet.require("./lib/message");
const serverActions = qu4rtet.require("./actions/serversettings");
const base64 = qu4rtet.require("base-64");
const {Parser} = qu4rtet.require("json2csv");
const jsonToXML = qu4rtet.require("jsontoxml");
const {pluginRegistry} = qu4rtet.require("./plugins/pluginRegistration");
import {setServerState} from "lib/reducer-helper";

export const initialData = () => ({
    servers: {},
    region: {},
    currentRegions: []
});

export const loadResponseRules = async (server, response) => {
    try {
        sessionStorage.setItem("loadingRR", true);
        let responseRules = await pluginRegistry
            .getServer(server.serverID)
            .fetchListAll("serialbox_response_rules_list", {}, []);
        // poor-man's matching alg.
        let poolsMap = {};
        response.results.forEach(pool => {
            poolsMap[pool.id] = pool;
            pool.response_rules = [];
        });
        if (responseRules && responseRules.length > 0) {
            responseRules.forEach(responseRule => {
                try {
                    poolsMap[responseRule.pool].response_rules.push(responseRule);
                } catch (e) {
                    // ignore responseRule that don't have a pool set.
                    console.log("A response rule appears to not have a pool/rule assigned.");
                    console.log(e);
                }
            });
        }
        if (response) {
            setTimeout(()=> {
                sessionStorage.setItem("loadingRR", false);
              }, []);
            return response
        }
        console.log(response)
        return response
    } catch (e) {
        return response
    }
};

export const loadPools = server => {
    return dispatch => {
        getPools(server).then(async pools => {
            // load response rules, if available.
            pools = await loadResponseRules(server, pools);
            dispatch({
                type: actions.loadPools,
                payload: {
                    [server.serverID]: {pools: pools, server: server}
                }
            });
        });
    };
};

export const getResponseRules = pool => {
    return dispatch => {

    }
}

export const loadRegion = (server, regionName) => {
    return dispatch => {
        getRegion(server, regionName).then(region => {
            dispatch({
                type: actions.loadRegion,
                payload: region
            });
        });
    };
};

export const loadRegions = (server, pool) => {
    return dispatch => {
        // first get all pools again to refresh pool list.
        getPools(server)
            .then(async pools => {
                pools = await loadResponseRules(server, pools);
                dispatch({
                    type: actions.loadPools,
                    payload: {
                        serverID: server.serverID,
                        pools: pools,
                        server: server
                    }
                });

                // second get region for given pool (updated.)
                let updatedPool = pools.find(aPool => {
                    return aPool.machine_name === pool.machine_name;
                });
                getRegions(server, updatedPool).then(regions => {
                    dispatch({
                        type: actions.loadRegions,
                        payload: regions
                    });
                });
            })
            .catch(e => {
                showMessage({type: "error", msg: e});
            });
    };
};

export const deleteARegion = (server, pool, region) => {
    return dispatch => {
        deleteRegion(server, region)
            .then(response => {
                if (!response.ok) {
                    if (response.status === 403 || response.status === 401) {
                        pluginRegistry.getHistory().push("/access-denied");
                        return;
                    } else {
                        throw new Error(response);
                    }
                }
                if (response.ok && response.status === 204) {
                    dispatch(loadRegions(server, pool));
                    showMessage({
                        id: "plugins.numberRange.regionDeletedSuccessfully",
                        type: "warning"
                    });
                }
            })
            .catch(error => {
                showMessage({type: "error", msg: error.detail});
            });
    };
};

export const deleteAPool = (server, pool) => {
    return dispatch => {
        deletePool(server, pool)
            .then(response => {
                if (response.status === 403 || response.status === 401) {
                    pluginRegistry.getHistory().push("/access-denied");
                    return;
                }
                if (response.ok && response.status === 204) {
                    dispatch(loadPoolList(server));
                    showMessage({type: "warning", msg: "Pool deleted successfully"});
                }
            })
            .catch(error => {
                showMessage({type: "error", msg: error.detail});
            });
    };
};

export const deleteResponseRule = (server, responseRule, page) => {
    return dispatch => {
        pluginRegistry
            .getServer(server.serverID)
            .getClient()
            .then(client => {
                client.apis.serialbox
                    .serialbox_response_rules_delete(responseRule)
                    .then(result => {
                        return dispatch(loadPoolList(server, null, page, null));
                    })
                    .catch(e => {
                        showMessage({
                            type: "error",
                            msg: "An error occurred while attempting to delete response rule"
                        });
                    });
            });
    };
};

const generateFile = (server, pool, exportType, data, size) => {
    if (
        typeof data === "object" &&
        data.numbers &&
        typeof data.numbers === "string"
    ) {
        // parse array of numbers.
        data.numbers = JSON.parse(data.numbers.replace(/'/g, '"'));
    }
    let encodedResult = "";
    // download the result.
    if (exportType === "json") {
        encodedResult = base64.encode(JSON.stringify(data));
    } else if (exportType === "csv") {
        let csvParser = new Parser();
        let numberMap = data.numbers.map(number => {
            return {
                numbers: number
            };
        });
        const csv = csvParser.parse(numberMap);
        encodedResult = base64.encode(csv);
    } else if (exportType === "xml") {
        encodedResult = base64.encode(data);
    }
    let link = document.createElement("a");
    let regionName = data.region || 0;
    link.download = `${pool.machine_name}-${regionName}-${size}.${exportType}`;
    link.href = `data:application/octet-stream;charset=utf-8;content-disposition:attachment;base64,${encodedResult}`;
    link.click();
};
export const setAllocation = (server, pool, value, exportType) => {
    return dispatch => {
        allocate(server, pool, value, exportType).then(data => {
            console.info('dispatch called...')
            if (typeof data === "object") {
                // let's take a look at the data.
                if (data.detail) {
                    // looks like an error.
                    showMessage({type: "error", msg: data.detail});
                } else if (data.fulfilled === true) {
                    showMessage({
                        id: "plugins.numberRange.allocatedSuccess",
                        type: "success",
                        values: {size: data.size_granted, regionName: data.region}
                    });
                }
            } else {
                showMessage({
                    id: "plugins.numberRange.allocatedSuccess",
                    type: "success",
                    values: {size: value, regionName: ""}
                });
            }
            dispatch({type: actions.allocate, payload: data});
            // reload regions.
            getRegions(server, pool).then(regions => {
                dispatch({
                    type: actions.loadRegions,
                    payload: regions
                });
            });
            setTimeout(() => {
                try {
                    if (data.size_granted || typeof data === "string") {
                        generateFile(server, pool, exportType, data, value);
                    }
                } catch (error) {
                    showMessage({
                        id: "plugins.numberRange.errorFailedToGenerateFile",
                        type: "danger"
                    });
                }
            }, 500);
        });
    };
};

export default handleActions(
    {
        [actions.loadPools]: (state, action) => {
            return setServerState(state, action.payload.serverID, {
                server: action.payload.server,
                pools: action.payload.pools,
                count: action.payload.count,
                next: action.payload.next
            });
        },
        [actions.loadRegion]: (state, action) => {
            return {
                ...state,
                region: action.payload
            };
        },
        [actions.loadRegions]: (state, action) => {
            return {
                ...state,
                currentRegions: action.payload
            };
        },
        [actions.allocate]: (state, action) => {
            // reload pools after
            //action.asyncDispatch(loadPools(action.payload));
            // don't pass allocation since it fails in redux.
            return {
                ...state
            };
        },

        [serverActions.serverUpdated]: (state, action) => {
            // we want to reload pools when new server is saved.
            /*action.asyncDispatch(
              loadPools(pluginRegistry.getServer(action.payload))
            );*/
            return {
                ...state
            };
        }
    },
    {}
);


export const loadPoolList = (server, search, page, ordering) => {
    const params = {};
    if (search) {
        params.search = search;
    }
    if (page) {
        params.page = page;
    }
    if (ordering) {
        params.ordering = ordering;
    }
    return async dispatch => {
        let serverObject = pluginRegistry.getServer(server.serverID);
        let response_pools = null;
        let pools = null;
        serverObject
            .fetchPageList("serialbox_pools_list", params, [])
            .then(async response => {
                response = await loadResponseRules(server, response);
                return dispatch({
                    type: actions.loadPools,
                    payload: {
                        serverID: server.serverID,
                        server: server,
                        pools: response.results,
                        count: response.count,
                        next: response.next
                    }
                })
            })
            .catch(e => {
                showMessage({
                    type: "error",
                    id: "plugins.masterData.errorFetchPools",
                    values: {error: e}
                });
            });
    };
};
