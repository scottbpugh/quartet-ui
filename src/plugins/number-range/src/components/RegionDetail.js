// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React, {Component} from "react";
import {connect} from "react-redux";
import {loadRegions} from "../reducers/numberrange";
import {Panels, RightPanel, LeftPanel} from "components/layouts/Panels";
import {Card, Callout} from "@blueprintjs/core";
import RegionRange from "./RegionRange";
import {setAllocation} from "../reducers/numberrange";
import "../style.css";
import classNames from "classnames";
import {FormattedDate, FormattedMessage, FormattedNumber} from "react-intl";

/**
 * _RegionDetail - Description
 * @extends Component
 */
class _RegionDetail extends Component {
  /*static propTypes = {
    history: PropTypes.object.isRequired
  };*/
  constructor(props) {
    super(props);
    this.state = {alloc: 1, lastUpdated: null};
    // these two properties below make it easy to retrieve
    // and trigger actions.
    this.currentPool = {readable_name: ""};
    this.currentServer = {};
  }
  componentDidMount() {
    this.loadPoolDetail(this.props);
  }
  previewAlloc = evt => {
    this.setState({alloc: Number(evt.target.value)});
  };
  loadPoolDetail(props) {
    let nrServer = props.nr[props.match.params.serverID];
    this.currentServer = nrServer.server;
    for (let pool of nrServer.pools) {
      if (pool.machine_name === props.match.params.pool) {
        this.currentPool = pool;
      }
    }
    this.props.loadRegions(this.currentServer, this.currentPool);
  }
  setAllocation = evt => {
    this.props.setAllocation(
      this.currentServer,
      this.currentPool,
      this.state.alloc
    );
  };
  componentWillReceiveProps(nextProps) {
    if (nextProps.match.params.pool !== this.props.match.params.pool) {
      this.loadPoolDetail(nextProps);
      return;
    }
    // shake card for the last updated region.
    /*nextProps.currentRegions.map((region, index) => {
      if (
        this.props.currentRegions[index] &&
        region.state !== this.props.currentRegions[index].state
      ) {
        console.log(
          "new state",
          region.state,
          "old state",
          this.props.currentRegions[index].state
        );
        this.setState({lastUpdated: region.machine_name}, () => {
          window.setTimeout(() => {
            this.setState({lastUpdated: null});
          }, 3000);
        });
      }
      return null;
    });*/
  }

  render() {
    let regions = this.props.currentRegions;

    return (
      <RightPanel
        title={
          <FormattedMessage
            id="plugins.numberRange.regionDetailTitle"
            values={{poolName: this.currentPool.readable_name}}
          />
        }>
        <div className="auto-cards-container">
          {regions && regions.length > 0 ? (
            regions.map(region => (
              <Card
                className={classNames({
                  "pt-elevation-4": true,
                  "region-detail": true,
                  updated: this.state.lastUpdated === region.machine_name
                })}
                key={region.machine_name}>
                <h5>{region.readable_name}</h5>
                <ul>
                  <li>
                    <FormattedMessage id="plugins.numberRange.createdOn" />:{" "}
                    <FormattedDate value={region.created_date} />
                  </li>
                  <li>
                    <FormattedMessage id="plugins.numberRange.status" />:{" "}
                    {region.active ? (
                      <FormattedMessage id="plugins.numberRange.active" />
                    ) : (
                      <FormattedMessage id="plugins.numberRange.inactive" />
                    )}
                  </li>
                  <li>
                    <FormattedMessage
                      id="plugins.numberRange.range"
                      defaultMessage="Range"
                    />: <FormattedNumber value={region.start} />{" "}
                    <FormattedMessage id="plugins.numberRange.to" />{" "}
                    <FormattedNumber value={region.end} />
                  </li>
                  <li>
                    <FormattedMessage id="plugins.numberRange.state" />:{" "}
                    <FormattedNumber value={region.state} />
                  </li>
                </ul>
                <RegionRange
                  start={region.start}
                  end={region.end}
                  state={region.state}
                  alloc={this.state.alloc}
                />
              </Card>
            ))
          ) : (
            <Callout>
              <FormattedMessage id="plugins.numberRange.noRegionInPool" />
            </Callout>
          )}
        </div>
      </RightPanel>
    );
    //<div>{JSON.stringify(this.props.region)}</div>;
  }
}

export var RegionDetail = connect(
  (state, ownProps) => {
    return {
      servers: state.serversettings.servers,
      currentRegions: state.numberrange.currentRegions,
      nr: state.numberrange.servers
    };
  },
  {loadRegions, setAllocation}
)(_RegionDetail);
