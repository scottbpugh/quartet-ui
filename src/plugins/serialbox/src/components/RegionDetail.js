// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {RegionCard} from "./RegionCard";
import "../style.css";
const React = qu4rtet.require("react");
const {Component} = React;
const {connect} = qu4rtet.require("react-redux");
const {pluginRegistry} = qu4rtet.require("./plugins/pluginRegistration");
const {withRouter} = qu4rtet.require("react-router");
const {Callout} = qu4rtet.require("@blueprintjs/core");
const {FormattedMessage} = qu4rtet.require("react-intl");
const classNames = qu4rtet.require("classnames");
const {DeleteDialog} = qu4rtet.require("./components/elements/DeleteDialog");
const {RightPanel} = qu4rtet.require("./components/layouts/Panels");
import {loadPools, loadRegions} from "../reducers/numberrange";

/**
 * _RegionDetail - Description
 * @extends Component
 */
class _RegionDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {alloc: 1, lastUpdated: null};
    // these two properties below make it easy to retrieve
    // and trigger actions.
    this.currentPool = {readable_name: "", machine_name: ""};
  }
  componentDidMount() {
    //this.props.loadPools(this.props.servers[this.props.match.params.serverID]);
    this.loadPoolDetail(this.props);
  }
  previewAlloc = evt => {
    this.setState({alloc: Number(evt.target.value)});
  };
  loadPoolDetail(props) {
    for (let pool of this.props.pools) {
      if (pool.machine_name === props.match.params.pool) {
        this.currentPool = pool;
      }
    }
    this.props.loadRegions(
      pluginRegistry.getServer(props.server.serverID),
      this.currentPool
    );
  }

  render() {
    let regions = this.props.currentRegions;

    return (
      <RightPanel
        title={
          <FormattedMessage
            id="plugins.numberRange.regionDetailTitle"
            values={{poolName: this.currentPool.readable_name}}
          />
        }>
        <div className="auto-cards-container">
          {regions && regions.length > 0 ? (
            regions.map(region => (
              <RegionCard
                key={region.machine_name}
                lastUpdated={this.state.lastUpdated}
                region={region}
                alloc={this.state.alloc}
                pool={this.currentPool}
                serverID={this.props.server.serverID}
                serverObject={pluginRegistry.getServer(
                  this.props.server.serverID
                )}
                history={this.props.history}
              />
            ))
          ) : (
            <Callout>
              <div className="pt-running-text">
                <FormattedMessage id="plugins.numberRange.noRegionInPool" />
              </div>
            </Callout>
          )}
        </div>
      </RightPanel>
    );
    //<div>{JSON.stringify(this.props.region)}</div>;
  }
}

export var RegionDetail = connect(
  (state, ownProps) => {
    return {
      server: state.serversettings.servers[ownProps.match.params.serverID],
      pools: state.numberrange.servers[ownProps.match.params.serverID].pools,
      currentRegions: state.numberrange.currentRegions
    };
  },
  {loadPools, loadRegions}
)(_RegionDetail);
