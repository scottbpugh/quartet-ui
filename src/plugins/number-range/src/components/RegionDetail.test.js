// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React, {Component} from "react";
import renderer from "react-test-renderer";
import {RegionDetail} from "./RegionDetail";
import {initialData} from "../../../../reducers/serversettings";
import {initialData as initialDataNR} from "../reducers/numberrange";
import configureStore from "redux-mock-store";
import {MemoryRouter as Router, withRouter, Switch} from "react-router-dom";
import {IntlProvider} from "react-intl";
import returnComponentWithIntl from "../../../../tools/intl-test-helper";
import thunk from "redux-thunk";

const middlewares = [thunk]; // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);
let wrapper;
let store;

it("renders correctly a pool with no region", () => {
  store = mockStore({
    serversettings: initialData(),
    numberrange: {
      servers: {
        fakeid: {
          server: {
            serverID: "fakeid",
            password: "toor",
            username: "root",
            port: "80",
            serverName: "localhost",
            serverSettingName: "fake server",
            ssl: false,
            path: ""
          },
          pools: [
            {
              sequentialregion_set: [],
              created_date: "2018-02-12T14:56:11.462232Z",
              modified_date: "2018-02-12T14:56:11.462279Z",
              readable_name: "Fake Pool",
              machine_name: "fakepool",
              active: true,
              request_threshold: 50000
            }
          ]
        }
      },
      region: {},
      currentRegions: []
    }
  });
  const props = {
    match: {
      params: {
        serverID: "fakeid"
      }
    }
  };
  window.fetch = () => {
    console.log("called");
  };
  const regionDetailScreen = renderer
    .create(
      returnComponentWithIntl(
        <Router initialEntries={["/number-range/region-detail/fakeid/blah/"]}>
          <RegionDetail
            store={store}
            match={{params: {serverID: "fakeid", pool: "fakepool"}}}
          />
        </Router>
      )
    )
    .toJSON();
  expect(regionDetailScreen).toMatchSnapshot();
});
