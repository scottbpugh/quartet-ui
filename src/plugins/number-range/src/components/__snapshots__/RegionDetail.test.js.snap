// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="left-panel pt-dark"
  >
    <h4
      className="left-panel-title"
    >
      Pool  Regions
    </h4>
    <div>
      <div
        className="mini-form"
      >
        <h6>
          <span>
            Allocation
          </span>
        </h6>
        <input
          className="pt-input"
          defaultValue={1}
          max={100000}
          min={1}
          onChange={[Function]}
          placeholder="allocate"
          style={
            Object {
              "width": 200,
            }
          }
          type="number"
        />
        <button
          className="pt-button"
          onClick={[Function]}
        >
          <span>
            Allocate from Pool
          </span>
        </button>
      </div>
      <div
        className="mini-form"
      >
        <button
          className="pt-button"
          onClick={[Function]}
        >
          <span>
            Add a New Region
          </span>
        </button>
      </div>
      <div
        className="pt-tree"
      >
        <ul
          className="pt-tree-node-list pt-tree-root"
        >
          <li
            className="pt-tree-node"
          >
            <div
              className="pt-tree-node-content pt-tree-node-content-0"
              onClick={[Function]}
              onContextMenu={[Function]}
              onDoubleClick={[Function]}
            >
              <span
                className="pt-tree-node-caret pt-icon-standard pt-tree-node-caret-closed"
                onClick={[Function]}
              />
              <span
                className="pt-tree-node-label"
              >
                fake server
              </span>
            </div>
            <div
              className="pt-collapse"
              style={
                Object {
                  "height": undefined,
                  "overflowY": undefined,
                  "transition": undefined,
                }
              }
            >
              <div
                aria-hidden={false}
                className="pt-collapse-body"
                style={
                  Object {
                    "transform": "translateY(-0px)",
                    "transition": undefined,
                  }
                }
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="right-panel"
  >
    <div>
      <div
        className="auto-cards-container"
      />
    </div>
  </div>
</div>
`;
